chain/indexer/src/streamer/streamer.rs

```diff
@@ -216,6 +238,7 @@ pub(crate) async fn start(
 
         let latest_block_height = block.header.height;
         if last_synced_block_height == 0 {
+            db.put(b"last_synced_block_height", &latest_block_height.to_string()).unwrap();
```

Comment:
There will be a bug if a user passes `BlockHeight(0)` or `BlockHeight(1)` as instead of syncing from block height 0 it will start with the latest while `BlockHeight(2)` will work correctly. I suggest you use `Option<_>` for `last_synced_block_height`.